### ISSUE TITLE
Navien integration fails in 2024.2

### ISSUE THREAD

**Issue by megaz555** [nikshriv/hass_navien_water_heater#35]
Hello,

Just wanted to point out a serious incompatibility in 2024.2 (beta ATM, but will be public soon). Once update is done, it is impossible to finish adding integration (times out after the polling screen). The integration also does not load properly when HA is restarted.

Here is the summary log:

```
Logger: custom_components.navien_water_heater.navien_api
Source: custom_components/navien_water_heater/navien_api.py:57
Integration: Navien NaviLink Water Heater Integration ([documentation](https://www.github.com/nikshriv/hass_navien_water_heater))
First occurred: 12:44:57 PM (15 occurrences)
Last logged: 12:48:50 PM

Connection error during start up: module 'ssl' has no attribute 'wrap_socket'
```

And here is the detailed log:

```
Traceback (most recent call last):
  File "/config/custom_components/navien_water_heater/navien_api.py", line 52, in start
    await self.login()
  File "/config/custom_components/navien_water_heater/navien_api.py", line 108, in login
    return await self._get_device_list()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/navien_water_heater/navien_api.py", line 129, in _get_device_list
    await self._connect_aws_mqtt()
  File "/config/custom_components/navien_water_heater/navien_api.py", line 150, in _connect_aws_mqtt
    await self.loop.run_in_executor(None,self.client.connect)
  File "/usr/local/lib/python3.12/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/AWSIoTPythonSDK/MQTTLib.py", line 520, in connect
    return self._mqtt_core.connect(keepAliveIntervalSecond)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/AWSIoTPythonSDK/core/protocol/mqtt_core.py", line 196, in connect
    self.connect_async(keep_alive_sec, self._create_blocking_ack_callback(event))
  File "/usr/local/lib/python3.12/site-packages/AWSIoTPythonSDK/core/protocol/mqtt_core.py", line 223, in connect_async
    raise e
  File "/usr/local/lib/python3.12/site-packages/AWSIoTPythonSDK/core/protocol/mqtt_core.py", line 211, in connect_async
    rc = self._internal_async_client.connect(keep_alive_sec, ack_callback)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/AWSIoTPythonSDK/core/protocol/internal/clients.py", line 125, in connect
    rc = self._paho_client.connect(host, port, keep_alive_sec)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/AWSIoTPythonSDK/core/protocol/paho/client.py", line 665, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/AWSIoTPythonSDK/core/protocol/paho/client.py", line 800, in reconnect
    rawSSL = ssl.wrap_socket(sock, ca_certs=self._tls_ca_certs, cert_reqs=ssl.CERT_REQUIRED)  # Add server certificate verification
             ^^^^^^^^^^^^^^^
AttributeError: module 'ssl' has no attribute 'wrap_socket'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 442, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/navien_water_heater/__init__.py", line 25, in async_setup_entry
    await navilink.start()
  File "/config/custom_components/navien_water_heater/navien_api.py", line 58, in start
    await asyncio.sleep(15)
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 665, in sleep
    return await future
           ^^^^^^^^^^^^
asyncio.exceptions.CancelledError: Global task timeout
```

**Comment by bakerkj** [comment]
Alternatively fixing this might require this integration upgrade from [AWSIoTPythonSDK](https://github.com/aws/aws-iot-device-sdk-python) to the rewrite of that library [aws-iot-device-sdk-python-v2](https://github.com/aws/aws-iot-device-sdk-python-v2).

**Comment by richie510** [comment]
I'm just adding that this integration broke for me when updating to 2024.2. I gained more than I lost (with 2024.2), but I would really like to keep this integration going. Looking at the history of this repository and the recent activity from @nikshriv I'm worried that this integration may be stuck @bakerkj do you think you can make the necessary changes described? If there is not response from @nikshriv will this have to be forked? I'm not a great programmer, but I can try really hard when needed, and I really want this integration to work again.

**Comment by plipof** [comment]
Same here. I have a Navien NFC-250H Combi-Boiler and will help any way I can.

**Comment by screwball571** [comment]
Same issue for me when I upgraded to 2024.2

**Comment by bakerkj** [comment]
I haven't had a chance to dig into writing a PR to move this integration from using the v2 AWSIoTSDK api. However there is a new PR up to fix the existing v1 API to work under Python 3.12 (https://github.com/aws/aws-iot-device-sdk-python/pull/338). Hopefully that will get reviewed, merged, and a release generated soon!

**Comment by posicat** [comment]
I'm running into the same, I'm open to help in any way I can.

**Comment by bakerkj** [comment]
As I noted the maintainers of AWSIoSDK fixed the issue with the SDK running under Python 3.12 in this release https://github.com/aws/aws-iot-device-sdk-python/releases/tag/v1.5.4. The new dependency is available with pip. 

I put up a PR (https://github.com/nikshriv/hass_navien_water_heater/pull/37) to require that version of the SDK in this Navien integration.

I hope @nikshriv can integrate this change into this repo. Until then if anyone would like to use my branch it is available here (https://github.com/bakerkj/hass_navien_water_heater).

**Comment by tayloracarr** [comment]
> As I noted the maintainers of AWSIoSDK fixed the issue with the SDK running under Python 3.12 in this release https://github.com/aws/aws-iot-device-sdk-python/releases/tag/v1.5.4. The new dependency is available with pip.
> 
> I put up a PR (#37) to require that version of the SDK in this Navien integration.
> 
> I hope @nikshriv can integrate this change into this repo. Until then if anyone would like to use my branch it is available here (https://github.com/bakerkj/hass_navien_water_heater).

Thank you!!!!

**Comment by kchiem** [comment]
> I hope @nikshriv can integrate this change into this repo. Until then if anyone would like to use my branch it is available here (https://github.com/bakerkj/hass_navien_water_heater).

I just upgraded to 2024.2.1 last night without changing repositories, and HA picked up the 1.5.4 version of AWSIoTPythonSDK.

**Comment by bakerkj** [comment]
> > I hope @nikshriv can integrate this change into this repo. Until then if anyone would like to use my branch it is available here (https://github.com/bakerkj/hass_navien_water_heater).
> 
> I just upgraded to 2024.2.1 last night without changing repositories, and HA picked up the 1.5.4 version of AWSIoTPythonSDK.

Great!

**Comment by bakerkj** [comment]
@nikshriv I think this can be closed now.
